tests/ExampleTests.cpp

```diff
@@ -65,6 +75,52 @@ TEST_SUITE("Example Tests") {
         CHECK(item > 0);
     }
 
+    // ----------------------------------------------
+    //             YAML Config Examples
+    // ----------------------------------------------
+
+    std::string transmogrifier(const std::string& s) {
+        // an example function with differing output for some scenarios
+        if (s == "scenario2") {
+            return "bar";
+        }
+        return "foo";
+    }
+
+    void checkAgainstExpectedResults(const std::string& scenarioName, YAML::Node expectedResults) {
+
+        // we have to specify the type of the desired data from the config via as()
+        CHECK(3.14159 == expectedResults["pi"].as<double>());
+        CHECK(2 == expectedResults["onlyEvenPrime"].as<int>());
+
+        // as() does work for some non-scalar types, such as vectors, lists, and maps
+        // for adding custom types to the config, see:
+        // https://github.com/jbeder/yaml-cpp/wiki/Tutorial#converting-tofrom-native-data-types
+        auto fib = expectedResults["fibonacciSeq"].as<std::vector<int>>();
```

Comment:
```suggestion
        const auto& fib = expectedResults["fibonacciSeq"].as<std::vector<int>>();
```

Reply:
Actually ignore the `&` on my suggestion here. I didn't realize that `as` doesn't return a reference.